### PR TITLE
Add graphs for table migration section

### PR DESCRIPTION
**Pull request by JonasJ-ap**
Add graphs for the introduction to table migration section in https://github.com/apache/iceberg/pull/6600

### CODE REVIEW

**Comment by JonasJ-ap** [comment]
I made up two html preview here:

https://drive.google.com/drive/folders/1XNZ1s2rXNSUxpP_ccUBVockV5KZ3gMOG?usp=share_link
(Since github does not allow me to upload html files directly)

File: iceberg-theme/static/img/iceberg-in-place-migration.png
Comment (jackye1995) [inline iceberg-theme/static/img/iceberg-in-place-migration.png:1]: I think we should specify that migrated table is "migrated Iceberg table", with "Iceberg Metadata" that has other things like manifest list, manifests, etc.

File: iceberg-theme/static/img/iceberg-snapshotaction-step1.png
Comment (jackye1995) [inline iceberg-theme/static/img/iceberg-snapshotaction-step1.png:1]: we should avoid using AWS logos for OSS doc

File: iceberg-theme/static/img/iceberg-snapshotaction-step2.png
[the remark this answers is not in the document]
Comment (jackye1995) [reply iceberg-theme/static/img/iceberg-snapshotaction-step2.png:1]: Also I think we can move the readers to the migrated table, showing users can switch readers and keep the writers.

**Review by jackye1995 — APPROVED** [review]
looks good to me!

**Comment by jackye1995** [comment]
Merging as a part of https://github.com/apache/iceberg/pull/6600

**Pull request merged**